thankyou_letters/models/partner_communication.py

```diff
@@ -33,6 +33,21 @@ class PartnerCommunication(models.Model):
         domain=[('type', '=', 'sentence')])
     success_sentence = fields.Text(related='success_sentence_id.body_text')
     add_success_story = fields.Boolean(related='config_id.add_success_story')
+    amount = fields.Float(string='Amount', compute='_compute_donation_amount',
+                          store=True)
+
+    @api.multi
+    def _compute_donation_amount(self):
+        for elem in self:
+            object_ids = map(int, elem.object_ids.split(','))
+            model = elem.config_id.model
+            amount_invoice = 0
+            for invoice_line_id in object_ids:
```

Comment:
The for loop is not necessary. You can treat directly all invoice_lines at one (see my following comments)